src/masternodes/mn_checks.cpp

```diff
@@ -3788,6 +3800,64 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
         return mnview.AddProposalVote(obj.propId, obj.masternodeId, vote);
     }
 
+    Res operator()(const CTransferBalanceMessage &obj) const {
+        auto res = Res::Ok();
+
+        // owner auth
+        if (obj.type != CTransferBalanceType::EvmOut)
```

Comment:
why don't we need to check auth for EvmOut too? Is that check happening on EVM side? Without the check, a transferbalance EvmOut might currently just create DFI out of thin air, right?